.github/workflows/ci.yml

```diff
@@ -29,92 +28,59 @@ jobs:
       - name: typos
         uses: crate-ci/typos@master
 
-      - name: setup uv
-        uses: astral-sh/setup-uv@v4
-        with:
-          enable-cache: true
-          python-version: "3.13"
-
-      - name: install
-        run: uv sync
-
-      - name: repo-review
-        run: uv run repo-review .
-
-      - name: mdformat
-        run: uv run mdformat --check
-
-      - name: ruff
-        run: |
-          uv run ruff check --output-format=github
-          uv run ruff format --check
-
-  typetest:
+  generate-matrix:
+    name: Generate Test Matrix
     timeout-minutes: 5
     runs-on: ubuntu-latest
+
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
     steps:
       - uses: actions/checkout@v4
 
-      - name: setup uv
+      - name: Install uv
         uses: astral-sh/setup-uv@v4
-        with:
-          enable-cache: true
-          python-version: "3.13"
 
-      - name: install
-        run: uv sync --no-editable
+      - name: Install dependencies
+        run: |
+          uv sync --frozen
+          uv pip install packaging
 
-      - name: basedpyright
-        run: >
-          uv run --no-sync
-          basedpyright
+      - name: Generate Matrix
+        id: set-matrix
+        shell: bash
+        run: |
+          matrix=$(uv run scripts/generate_matrix.py | jq -c)
+          echo "matrix=$matrix" >> $GITHUB_OUTPUT
```

Comment:
Maybe `uv pip install packaging` could be replaced with `uv add --script`?
https://docs.astral.sh/uv/guides/scripts/#declaring-script-dependencies

or alternatively, `uv run --with packaging ...`?